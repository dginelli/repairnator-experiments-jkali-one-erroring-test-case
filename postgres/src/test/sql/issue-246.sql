create table issue246 (
  id serial8,
  data json
);

create index idxissue246 on issue246 using zombodb (zdb('issue246', ctid), zdb(issue246)) with (url='localhost:9200/');

insert into issue246 (data) values ('[{"id":1, "state_id": 42},{"id":2, "state_id": 42}]');
insert into issue246 (data) values ('[{"id":3, "state_id": 66},{"id":4, "state_id": 42},{"id":5, "state_id": 66}]');
insert into issue246 (data) values ('[{"id":6, "state_id": [42,66]}]');
insert into issue246 (data) values ('[{"id":7, "state_id": [42,66]},{"id":8, "state_id": 75}]');

select * from issue246 order by id;
select * from issue246 where zdb('issue246', ctid) ==> 'data.state_id = 42 and data.state_id = 66' order by id;
select * from issue246 where zdb('issue246', ctid) ==> 'data.state_id = 42 with data.state_id = 66' order by id;

select * from zdb_tally('issue246', 'data.id', true, '^.*', '', 5000, 'term');
select * from zdb_tally('issue246', 'data.id', true, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
select * from zdb_tally('issue246', 'data.id', true, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');

select * from zdb_tally('issue246', 'data.id', false, '^.*', '', 5000, 'term');
select * from zdb_tally('issue246', 'data.id', false, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
select * from zdb_tally('issue246', 'data.id', false, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');

select * from zdb_tally('issue246', 'id', false, '^.*', '', 5000, 'term');
select * from zdb_tally('issue246', 'id', false, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
select * from zdb_tally('issue246', 'id', false, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');


drop table issue246 cascade;