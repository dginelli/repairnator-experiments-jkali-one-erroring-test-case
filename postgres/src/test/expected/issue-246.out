create table issue246 (
  id serial8,
  data json
);
create index idxissue246 on issue246 using zombodb (zdb('issue246', ctid), zdb(issue246)) with (url='localhost:9200/');
insert into issue246 (data) values ('[{"id":1, "state_id": 42},{"id":2, "state_id": 42}]');
insert into issue246 (data) values ('[{"id":3, "state_id": 66},{"id":4, "state_id": 42},{"id":5, "state_id": 66}]');
insert into issue246 (data) values ('[{"id":6, "state_id": [42,66]}]');
insert into issue246 (data) values ('[{"id":7, "state_id": [42,66]},{"id":8, "state_id": 75}]');
select * from issue246 order by id;
 id |                                     data                                     
----+------------------------------------------------------------------------------
  1 | [{"id":1, "state_id": 42},{"id":2, "state_id": 42}]
  2 | [{"id":3, "state_id": 66},{"id":4, "state_id": 42},{"id":5, "state_id": 66}]
  3 | [{"id":6, "state_id": [42,66]}]
  4 | [{"id":7, "state_id": [42,66]},{"id":8, "state_id": 75}]
(4 rows)

select * from issue246 where zdb('issue246', ctid) ==> 'data.state_id = 42 and data.state_id = 66' order by id;
 id |                                     data                                     
----+------------------------------------------------------------------------------
  2 | [{"id":3, "state_id": 66},{"id":4, "state_id": 42},{"id":5, "state_id": 66}]
  3 | [{"id":6, "state_id": [42,66]}]
  4 | [{"id":7, "state_id": [42,66]},{"id":8, "state_id": 75}]
(3 rows)

select * from issue246 where zdb('issue246', ctid) ==> 'data.state_id = 42 with data.state_id = 66' order by id;
 id |                           data                           
----+----------------------------------------------------------
  3 | [{"id":6, "state_id": [42,66]}]
  4 | [{"id":7, "state_id": [42,66]},{"id":8, "state_id": 75}]
(2 rows)

select * from zdb_tally('issue246', 'data.id', true, '^.*', '', 5000, 'term');
 term | count 
------+-------
 1    |     1
 2    |     1
 3    |     1
 4    |     1
 5    |     1
 6    |     1
 7    |     1
 8    |     1
(8 rows)

select * from zdb_tally('issue246', 'data.id', true, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
 term | count 
------+-------
 6    |     1
 7    |     1
(2 rows)

select * from zdb_tally('issue246', 'data.id', true, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');
 term | count 
------+-------
 6    |     1
 7    |     1
(2 rows)

select * from zdb_tally('issue246', 'data.id', false, '^.*', '', 5000, 'term');
 term | count 
------+-------
 1    |     1
 2    |     1
 3    |     1
 4    |     1
 5    |     1
 6    |     1
 7    |     1
 8    |     1
(8 rows)

select * from zdb_tally('issue246', 'data.id', false, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
 term | count 
------+-------
 3    |     1
 4    |     1
 5    |     1
 6    |     1
 7    |     1
 8    |     1
(6 rows)

select * from zdb_tally('issue246', 'data.id', false, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');
 term | count 
------+-------
 6    |     1
 7    |     1
 8    |     1
(3 rows)

select * from zdb_tally('issue246', 'id', false, '^.*', '', 5000, 'term');
 term | count 
------+-------
 1    |     1
 2    |     1
 3    |     1
 4    |     1
(4 rows)

select * from zdb_tally('issue246', 'id', false, '^.*', 'data.state_id=42 and data.state_id=66', 5000, 'term');
 term | count 
------+-------
 2    |     1
 3    |     1
 4    |     1
(3 rows)

select * from zdb_tally('issue246', 'id', false, '^.*', 'data.state_id=42 with data.state_id=66', 5000, 'term');
 term | count 
------+-------
 3    |     1
 4    |     1
(2 rows)

drop table issue246 cascade;
